### ISSUE TITLE
Add loose mode

### ISSUE THREAD

**Issue by jdalton** [jridgewell/babel-plugin-transform-for-of-as-array#2]
By spec nullish values will error out but that leads to having to do

```js
if (maybeNullish) {
  for (let v of maybeNullish) {}
}
```

It would be rad if there was a `loose` mode to have it do
`length = array ? array.length : 0` so I could just do

```js
for (let v of maybeNullish) {}
```

**Comment by jridgewell** [comment]
The plugin was [sherlocked](https://github.com/babel/babel/pull/6914) by babel. Are you still using it?

**Comment by jdalton** [comment]
Yep !

**Comment by jridgewell** [comment]
Want to write the PR?

**Comment by jdalton** [comment]
If you have bandwidth I'll defer to you, but if you don't I'll put it on my list to tackle :P

**Comment by jridgewell** [comment]
Fixed by https://github.com/jridgewell/babel-plugin-transform-for-of-as-array/commit/b7737b0230c5a3f292b9a52bbf40574f5f298d25, released as v1.1.0.

```js
// loose
for (let _i = 0, _array = array, _length = _array == null ? 0 : _array.length; _i < _length; _i++) {
  const elm = _array[_i];
  console.log(elm);
} 

// normal
for (let _i = 0, _array = array, _length = _array.length; _i < _length; _i++) {
  const elm = _array[_i];
  console.log(elm);
}

**Comment by jdalton** [comment]
Thank you!
